presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -220,13 +227,16 @@ private static Partitioner createPartitioner(PartitioningHandle partitioning, in
                 .flatMap(List::stream)
                 .collect(toImmutableSet());
 
-        Set<PlanFragmentId> missingInputs = difference(expectedInputs, rddInputs.keySet());
-        Set<PlanFragmentId> extraInputs = difference(rddInputs.keySet(), expectedInputs);
+        Set<PlanFragmentId> actualInputs = union(rddInputs.keySet(), broadcastInputs.keySet());
+
+        Set<PlanFragmentId> missingInputs = difference(expectedInputs, actualInputs);
+        Set<PlanFragmentId> extraInputs = difference(actualInputs, expectedInputs);
         checkArgument(
                 missingInputs.isEmpty() && extraInputs.isEmpty(),
```

Comment:
Curious: can we just do `actualInputs.equals(expectedInputs)`?

Reply:
I was going to add `missingInputs ` and `extraInputs ` to the error message, but then forgot. Let me do this. It might be convenient when debugging.

Reply:
@arhimondr : In that case maybe consider refactor the check logic into a separate method (e.g. `checkRddInputs` to keep the core logic straightforward? :)